monitoring/rid_qualifier/flight_data_generator.py

```diff
@@ -0,0 +1,481 @@
+from shapely.geometry import Point, Polygon, LineString
+import shapely.geometry
+from pyproj import Geod, Transformer, Proj
+import json
+from pathlib import Path
+from typing import List, NamedTuple, Any
+import arrow
+import datetime
+from datetime import datetime, timedelta
+
+
+class QueryBoundingBox(NamedTuple):
+    ''' This is the object that stores details of query bounding box '''
+
+    name: str
+    shape: Polygon
+    timestamp_before: timedelta
+    timestamp_after: timedelta
+
+
+class FlightPoint(NamedTuple):
+    ''' This object holds basic information about a point on the flight track, it has latitude, longitude and altitude in WGS 1984 datum '''
+
+    lat: float  # Degrees of latitude north of the equator, with reference to the WGS84 ellipsoid. For more information see: https://github.com/uastech/standards/blob/master/remoteid/canonical.yaml#L1160
+    lng: float  # Degrees of longitude east of the Prime Meridian, with reference to the WGS84 ellipsoid. For more information see: https://github.com/uastech/standards/blob/master/remoteid/canonical.yaml#L1170
+    alt: float  # meters in WGS 84, normally calculated as height of ground level in WGS84 and altitude above ground level
+    speed: float # speed in m / s 
+    bearing: float # forward azimuth for the this and the next point on the track
+
+
+class AircraftPosition(NamedTuple):
+    ''' A object to hold AircraftPosition details for Remote ID purposes, it mataches the RIDAircraftPosition  per the RID standard, for more information see https://github.com/uastech/standards/blob/36e7ea23a010ff91053f82ac4f6a9bfc698503f9/remoteid/canonical.yaml#L1091'''
+
+    lat: float
+    lng: float
+    alt: float
+    accuracy_h: str
+    accuracy_v: str
+    extrapolated: bool
+
+
+class AircraftHeight(NamedTuple):
+    ''' A object to hold relative altitude for the purposes of Remote ID. For more information see: https://github.com/uastech/standards/blob/36e7ea23a010ff91053f82ac4f6a9bfc698503f9/remoteid/canonical.yaml#L1142 '''
+
+    distance: float
+    reference: str
+
+
+class AircraftState(NamedTuple):
+    ''' A object to hold Aircraft state details for remote ID purposes. For more information see the published standard API specification at https://github.com/uastech/standards/blob/36e7ea23a010ff91053f82ac4f6a9bfc698503f9/remoteid/canonical.yaml#L1604 '''
+
+    timestamp: datetime
```

Comment:
For consistency with the RID spec, this type is a `str` rather than a `datetime` (and it is populated with a `str` on line 310)

Reply:
Thanks yes, just as a side question, Python is letting me insert string instead of date time, is there any way that we can do this type checking Python?